lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -769,7 +777,7 @@ Future<DomXMLHttpRequest> domHttpRequest(String url,
     }
   }));
 
-  xhr.addEventListener('error', allowInterop(completer.completeError));
+  xhr.addEventListener('error', allowInterop((DomEvent event) => completer.completeError(event)));
```

Comment:
I wonder if this is still necessary?

Reply:
err, actually best to just leave this alone for the time being, this, or something similar, might be why we are seeing the crash.

Reply:
I think we should be able to land without this change now, and once https://dart-review.googlesource.com/c/sdk/+/266869 lands the behavior between backends w.r.t. default arguments on callbacks should be identical.